pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java

```diff
@@ -495,5 +493,9 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         }
     }
 
+    private void writeAndFlush(ByteBuf cmd) {
+        NettyChannelUtil.writeAndFlushWithVoidPromise(outboundChannel, cmd);
```

Comment:
Do you miss `FIRE_EXCEPTION_ON_FAILURE`?

Reply:
@nodece VoidChannelPromise [will propagate exceptions](https://github.com/netty/netty/blob/0097c2b72521c422d42bf8c45273db05a7ff068c/transport/src/main/java/io/netty/channel/VoidChannelPromise.java#L177-L180).